middleware/kubernetes/lookup.go

```diff
@@ -0,0 +1,304 @@
+package kubernetes
+
+import (
+	"fmt"
+	"math"
+	"net"
+	"time"
+
+	"github.com/miekg/coredns/middleware"
+	"github.com/miekg/coredns/middleware/kubernetes/msg"
+
+	"github.com/miekg/dns"
+)
+
+func (k Kubernetes) records(state middleware.State, exact bool) ([]msg.Service, error) {
+	services, err := k.Records(state.Name(), exact)
+	if err != nil {
+		return nil, err
+	}
+	services = msg.Group(services)
```

Comment:
Probably not. I suspect that much (or all) of this lookup.go file will be deleted as k8s gets built out.
Added a TODO comment in this function to track this question.